code/modules/effects/map_effects/door_helper.dm

```diff
@@ -0,0 +1,12 @@
+/obj/effect/map_effect/door_helper
+	layer = DOOR_CLOSED_LAYER + 0.1
+
+/obj/effect/map_effect/door_helper/unres
+	icon_state = "unres_door"
+
+/obj/effect/map_effect/door_helper/unres/Initialize(mapload, ...)
+	..()
+	var/obj/machinery/door/D = locate() in loc
```

Comment:
If I have a blast door, and a regular door, in the same loc (as is common) this will find the first door and give it the unres_dir -- ignoring the other one.